ci/scripts/deterministic-recovery-test.sh

```diff
@@ -9,11 +9,19 @@ echo "--- Download artifacts"
 download-and-decompress-artifact risingwave_simulation .
 chmod +x ./risingwave_simulation
 
-export RUST_LOG="info,risingwave_meta::barrier::recovery=debug"
+export RUST_LOG="info,\
+risingwave_meta::barrier::recovery=debug,\
+risingwave_meta::rpc::ddl_controller=debug,\
+risingwave_meta::barrier::mod=debug,\
+risingwave_simulation=debug"
 export LOGDIR=.risingwave/log
 
 mkdir -p $LOGDIR
 
+# FIXME(kwannoel): Why is this failing?
+# echo "--- deterministic simulation e2e, ci-3cn-2fe-3meta, recovery, background_ddl"
+# seq $TEST_NUM | parallel MADSIM_TEST_SEED={} './risingwave_simulation --kill --kill-rate=${KILL_RATE} ./e2e_test/background_ddl/sim/basic.slt 2> $LOGDIR/recovery-ddl-{}.log && rm $LOGDIR/recovery-ddl-{}.log'
+
```

Comment:
Seems like under deterministic simulation, the ddl does not run in background. Not sure why.
Backfill tests cover the recovery scenarios instead.

Reply:
What does the error log looks like? You can paste it here to see if I can find some clues.

Reply:
There's no log, let me describe the issue in greater detail:
1. When running 3 different MVs, I would expect these MVs to run immediately run in the background.
2. Then, `rw_catalog.rw_ddl_progress` should show 3 jobs.
3. ~30s after we should get the test results when the MVs are created.

All ill all, this process should be pretty fast, maybe 3min. But after increasing the number of records to 100M, it appears to slow down, running over 14minutes.
Additionally, step 2. does not happen, we see the count as 0.

So it seems that the background ddl somehow blocks until mv has finished backfilling.

Reply:
After checking the details about the progress report for recovered jobs, didn't find any clues that will cause the count as 0. 🥵 Can you add more logs for the progress report and test locally to find the root case for it?

Reply:
Sharing the test case:
```
statement ok
SET BACKGROUND_DDL=true;

statement ok
ALTER SYSTEM SET max_concurrent_creating_streaming_jobs TO 4;

statement ok
CREATE TABLE t (v1 int, dummy int);

# This is expected to run under sim mode, which is fast,
# so we need a higher number of rows.
statement ok
INSERT INTO t select generate_series, 1 from generate_series(1, 1000000);

statement ok
FLUSH;

statement ok
CREATE MATERIALIZED VIEW m1 as SELECT * FROM t;

statement ok
CREATE MATERIALIZED VIEW m2 as SELECT * FROM t;

statement ok
CREATE MATERIALIZED VIEW m3 as SELECT * FROM t;
```

When running this in `deterministic-simulation-mode`, RW blocks for 4min
When running this normally via sqllogictest, RW is able to immediately return, within 38s.

Still investigating.

Reply:
Oh I think I figured it out. Madsim does not see BACKGROUND_DDL, because it has its own special handler... `sim/ddl_service`

Reply:
I will fix this in subsequent PRs.
I think the existing tests should sufficiently cover the cases.

I think there's 2 main issues here:
1. We should not test for `rw_progress`, seems to be flaky.
2. We should test for background_ddl to always be created.

Opened a tracking issue for it: https://github.com/risingwavelabs/risingwave/issues/12957.